flutter-gui-tests/testSrc/io/flutter/tests/gui/ProjectCreator.kt

```diff
@@ -1,60 +1,131 @@
 package io.flutter.tests.gui
 
+import com.intellij.ide.fileTemplates.impl.UrlUtil
 import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.util.io.StreamUtil
+import com.intellij.platform.templates.github.ZipUtil
 import com.intellij.testGuiFramework.framework.Timeouts
 import com.intellij.testGuiFramework.impl.*
 import com.intellij.testGuiFramework.utils.TestUtilsClass
 import com.intellij.testGuiFramework.utils.TestUtilsClassCompanion
+import com.intellij.util.UriUtil
+import com.intellij.util.io.URLUtil
 import org.fest.swing.exception.WaitTimedOutError
+import java.io.BufferedOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.net.URL
 
 val GuiTestCase.ProjectCreator by ProjectCreator
 
 // Inspired by CommunityProjectCreator
 class ProjectCreator(guiTestCase: GuiTestCase) : TestUtilsClass(guiTestCase) {
 
-  companion object : TestUtilsClassCompanion<ProjectCreator>({ it -> ProjectCreator(it) })
+  companion object : TestUtilsClassCompanion<ProjectCreator>({ ProjectCreator(it) })
 
   private val defaultProjectName = "untitled"
-  private val LOG = Logger.getInstance(this.javaClass)
+  private val sampleProjectName = "simple_app"
+  private val log = Logger.getInstance(this.javaClass)
+
+  fun importProject(projectName: String = sampleProjectName): File {
+    val projectDirFile = extractProject(projectName)
+    val projectPath: File = guiTestCase.guiTestRule.importProject(projectDirFile)
+    with(guiTestCase) {
+      openMainInProject()
+      waitForFirstIndexing()
+      ideFrame {
+        editor {
+          try {
+            val note = notificationPanel()
+            // Best practice is to avoid reusing strings defined in bundles.
+            if (note?.getLabelText() == "Pubspec has been edited") {
```

Comment:
I'd be interested to see if we ever hit race conditions here.  (In my experience there can be a lag in the banner appearing.) But maybe `waitForFirstIndexing()` normalizes this away?

Reply:
I wish. Sometimes the banner is delayed so long the catch is triggered. I suspect there is some way to cause all background tasks to complete but I haven't found it yet.